src/utils/errors/client.error.ts

```diff
@@ -43,6 +43,12 @@ export class FileExistsWithSameName extends ClientError {
   }
 }
 
+export class FileParentAppIDNotEqual extends ClientError {
+  constructor(message?: string) {
+    super(message || 'file parent app id is not the same', grpc.status.INVALID_ARGUMENT);
```

Comment:
The message is not clear enough. I think you should also add the different parent and file appIDs.

Reply:
Also, I'm not sure that `INVALID_ARGUMENT` is the appropriate error type. Maybe it should be `PERMISSION_DENIED`.

Reply:
done.